src/controllers/courseController.ts

```diff
@@ -23,15 +23,17 @@ export class CourseController {
 	}
 
 	public courseOverview() {
-		logger.debug('Loading Course Overview page')
-		return async (request: Request, response: Response) => {
-			const req = request as CourseRequest
-
-			const course = req.course
+		logger.debug('Course Overview page')
+		const self = this
 
-			response.render(`page/course`, {
-				course,
-			})
+		return async (request: Request, response: Response) => {
+			const courseId: string = request.params.courseId
+			const course = await self.learningCatalogue.getCourse(courseId)
```

Comment:
INFO - why is loadCourse public?
INFO - Loading a course or 404 logic is repeated (with template variation)